hudi-common/src/main/java/org/apache/hudi/common/util/AvroOrcUtils.java

```diff
@@ -511,8 +511,8 @@ public static Object readFromVector(TypeDescription type, ColumnVector colVector
           throw new HoodieIOException("CHAR/VARCHAR has length " + result.length() + " greater than Max Length allowed");
         }
       case STRING:
-        String stringType = avroSchema.getProp(GenericData.STRING_PROP);
-        if (stringType == null || !stringType.equals(StringType.String)) {
+        String stringType = avroSchema != null ? avroSchema.getProp(GenericData.STRING_PROP) : null;
+        if (!StringType.String.name().equals(stringType)) {
           int stringLength = ((BytesColumnVector) colVector).length[vectorPos];
```

Comment:
What about `stringType == null || !StringType.String.name().equals(stringType)` ?

Reply:
Check if user set type as string, default is utf8 type. 
```
  public enum StringType { CharSequence, String, Utf8 };
  ...
  public static final String STRING_PROP = "avro.java.string";
  protected static final String STRING_TYPE_STRING = "String";
  ...
  s.addProp(GenericData.STRING_PROP, GenericData.STRING_TYPE_STRING);
```
`StringType.String` is an enumeration type, but `stringType` is of type String.

Reply:
So, the original code `stringType == null` returns true and the new code also returns true, what's the difference here ?

Reply:
This condition is to check if the user has set the type to string, the default is utf8 type. But `stringType` is `String` type and `StringType.String` is enum type. Below is the relevant code:
```
  public enum StringType { CharSequence, String, Utf8 };

  public static final String STRING_PROP = "avro.java.string";
  protected static final String STRING_TYPE_STRING = "String";

 ...
 s.addProp(GenericData.STRING_PROP, GenericData.STRING_TYPE_STRING);
```

Reply:
Then `stringType == null || !StringType.String.name().equals(stringType)` should work as expected ~

Reply:
`if (stringType == null || !StringType.String.name().equals(stringType))` should be equivalent to `!StringType.String.name().equals(stringType)`